### PR TITLE
Unify tests (alternative from review)

### PR DESCRIPTION
**Pull request by lucafavatellawork**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -132,28 +132,23 @@ test_setup: check_python
...
-		-s \
-		--ignore=sparkl_cli/test/data \
-		--doctest-glob='*.rst' \
-		--timeout=$(PYTEST_TIMEOUT) \
-		$(PYTEST_EXTRA_ARGS) \
-		$($@_dirs)
+define run_tests
+$(PYTHON) -m pytest \
+	-s \
+	--ignore=sparkl_cli/test/data \
```
Comment (lucafavatellawork) [inline Makefile:138]: Would it be more readable to use `$(pytest_dir)/data` here?

File: Makefile [hunk 2]
```diff
@@ -132,28 +132,23 @@ test_setup: check_python
...
+	$(1)
+endef
+pytest_dir = sparkl_cli/test
+doctest_dir = docs
+.PHONY: pytest
+pytest: test_setup; $(call run_tests,$(pytest_dir))
+.PHONY: doctest
+doctest: test_setup; $(call run_tests,$(doctest_dir))
+.PHONY: test
+test: test_setup; $(call run_tests,$(pytest_dir) $(doctest_dir))
```
Comment (lucafavatellawork) [inline Makefile:151]: Would it be more readable to write `sparkl_cli/test docs` here and delete the variables `pytest_dir` and `doctest_dir`? (If so, the paths would become magic strings repeated - not connected structurally. Though they are so few that anything goes.)

File: Makefile (hunk 1 above)
Comment (miklosduma) [reply Makefile:138]: As a matter of fact this argument also only makes sense if the test targets include `sparkl_cli/test` but hey :D

File: Makefile (hunk 2 above)
Comment (miklosduma) [reply Makefile:151]: Let's go with your original approach. If someone messes up any of the variables and starts complaining you can at least say why don't you read the comments?

Comment (lucafavatellawork) [reply Makefile:151]: I went for this approach, and removing vars. Thanks.

**Pull request merged**